kernel/src/table_features/column_mapping.rs

```diff
@@ -52,15 +174,15 @@ mod tests {
 
         let protocol = Protocol::try_new(2, 5, None::<Vec<String>>, None::<Vec<String>>).unwrap();
         assert_eq!(
-            column_mapping_mode(&protocol, &table_properties),
+            column_mapping_mode(&protocol, &table_properties).unwrap(),
             ColumnMappingMode::Id
         );
 
         let empty_features = Some::<[String; 0]>([]);
         let protocol =
             Protocol::try_new(3, 7, empty_features.clone(), empty_features.clone()).unwrap();
         assert_eq!(
-            column_mapping_mode(&protocol, &table_properties),
+            column_mapping_mode(&protocol, &table_properties).unwrap(),
             ColumnMappingMode::None
```

Comment:
if there are no features with 3/7 protocol that implies no column mapping - but I think your is exposed as an error? (noticed failing test)